src/components/Application/Navigation.tsx

```diff
@@ -95,6 +95,15 @@ const Navigation: React.FC<NavigationProps> = ({
     })
   }
 
+  const summaryButtonHandler = () => {
+    requestRevalidation(({ firstStrictInvalidPage, setStrictSectionPage }: MethodToCallProps) => {
+      if (firstStrictInvalidPage) {
+        setStrictSectionPage(firstStrictInvalidPage)
+        sendToPage(firstStrictInvalidPage)
+      } else push(`/applicationNEW/${serialNumber}/summary`)
```

Comment:
if we restructure routing slightly, home, applicationPage and application summary, to have a wrapper that calls useGetFullApplicationStructure, then we can put the re-validation useEffect and the method there and pass down to summary, home and page. Then the logic in summary submit would be much simpler. What do you think ?

Reply:
ApplicationWrapper->
  ApplicationPageWrapper ->
      ApplicationPage
      ApplicationHome
      ApplicationSummary
Application Submit
...Review

Reply:
Righto, made a new branch based off this one with these suggested/discussed improvements:
https://github.com/openmsupply/application-manager-web-app/pull/424